tensorflow_lite_support/python/test/task/vision/image_segmenter_test.py

```diff
@@ -196,6 +231,85 @@ def test_segment_model(self, model_file_type, expected_colored_labels):
 
+  def test_segmentation_category_mask(self):
+    """Check if category mask matches with ground truth."""
+    # Creates segmenter.
+    base_options = _BaseOptions(file_name=self.model_path)
+    segmenter = _create_segmenter_from_options(
+      base_options, output_type=_OutputType.CATEGORY_MASK)
+
+    # Loads image.
+    image = tensor_image.TensorImage.create_from_file(self.test_image_path)
+
+    # Performs image segmentation on the input.
+    segmentation = segmenter.segment(image).segmentation[0]
+    result_pixels = np.array(bytearray(segmentation.category_mask))
+
+    # Loads ground truth segmentation file.
+    gt_segmentation = tensor_image.TensorImage.create_from_file(
+      self.test_seg_path)
+    gt_segmentation_array = gt_segmentation.buffer
+    gt_segmentation_shape = gt_segmentation_array.shape
+    num_pixels = gt_segmentation_shape[0] * gt_segmentation_shape[1]
+    ground_truth_pixels = gt_segmentation_array.flatten()
+
+    self.assertEqual(
+      len(result_pixels), len(ground_truth_pixels),
+      "Segmentation mask size does not match the ground truth mask size.")
+
+    inconsistent_pixels = 0
+
+    for index in range(num_pixels):
+      inconsistent_pixels += (
+          result_pixels[index] * _MASK_MAGNIFICATION_FACTOR !=
+          ground_truth_pixels[index])
+
+    self.assertLessEqual(
+      inconsistent_pixels / num_pixels, _MATCH_PIXELS_THRESHOLD,
+      f"Number of pixels in the candidate mask differing from that of the "
+      f"ground truth mask exceeds {_MATCH_PIXELS_THRESHOLD}.")
+
+  def test_segmentation_confidence_mask_matches_category_mask(self):
+    """Check if the confidence mask matches with the category mask."""
+    # Create BaseOptions from model file.
+    base_options = _BaseOptions(file_name=self.model_path)
+
+    # Loads image.
+    image = tensor_image.TensorImage.create_from_file(self.test_image_path)
+
+    # Run segmentation on the model in CATEGORY_MASK mode.
+    segmenter = _create_segmenter_from_options(
+      base_options, output_type=_OutputType.CATEGORY_MASK)
+
+    # Performs image segmentation on the input and gets the category mask.
+    segmentation = segmenter.segment(image).segmentation[0]
+    category_mask = np.array(bytearray(segmentation.category_mask))
+
+    # Run segmentation on the model in CONFIDENCE_MASK mode.
+    segmenter = _create_segmenter_from_options(
+      base_options, output_type=_OutputType.CONFIDENCE_MASK)
+
+    # Performs image segmentation on the input again.
+    segmentation = segmenter.segment(image).segmentation[0]
+
+    # Gets the list of confidence masks and colored_labels.
+    confidence_masks = segmentation.confidence_masks.confidence_mask
+    colored_labels = segmentation.colored_labels
+
+    # Check if confidence mask shape is correct.
+    self.assertEqual(len(confidence_masks), len(colored_labels),
+                     'Number of confidence masks must match with number of '
```

Comment:
Please don't mix single quote and double quote in the same file. Let's unify to double quote then.